pkg/serviceaccounts/controllers/create_dockercfg_secrets.go

```diff
@@ -81,8 +82,37 @@ func NewDockercfgController(cl client.Interface, options DockercfgControllerOpti
 			},
 		},
 	)
-
 	e.serviceAccountCache = NewEtcdMutationCache(serviceAccountCache)
+
+	tokenSecretSelector := fields.OneTermEqualSelector(api.SecretTypeField, string(api.SecretTypeServiceAccountToken))
+	e.secretsCache, e.secretController = framework.NewInformer(
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				opts := api.ListOptions{FieldSelector: tokenSecretSelector}
+				return e.client.Secrets(api.NamespaceAll).List(opts)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				opts := api.ListOptions{FieldSelector: tokenSecretSelector, ResourceVersion: options.ResourceVersion}
+				return e.client.Secrets(api.NamespaceAll).Watch(opts)
+			},
+		},
+		&api.Secret{},
+		options.Resync,
+		framework.ResourceEventHandlerFuncs{
+			AddFunc: func(cur interface{}) {
+				secret := cur.(*api.Secret)
+				glog.V(2).Infof("Adding token secret %s/%s", secret.Namespace, secret.Name)
+				e.handleTokenPopulation(nil, cur)
+			},
+			UpdateFunc: func(old, cur interface{}) {
+				secret := cur.(*api.Secret)
+				glog.V(2).Infof("Updating token secret %s/%s", secret.Namespace, secret.Name)
+				e.handleTokenPopulation(old, cur)
+			},
+			//DeleteFunc: e.handleTokenDeletion,
```

Comment:
you need to track deletions to know when we are no longer waiting for the token to be populated and rekick the sync of the service account